### PR TITLE
feat(cart): add in-memory drivers

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/cart/testing/src/drivers/in-memory/cart-shipping-address/cart-shipping-address.service.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export class DaffInMemoryCartShippingAddressService implements DaffCartShippingAddressServiceInterface<DaffCartAddress, DaffCart> {
+  url = '/api/cart-shipping-address';
+
+  constructor(private http: HttpClient) {}
+
+  get(cartId: DaffCart['id']): Observable<DaffCartAddress> {
+    return this.http.get<DaffCartAddress>(`${this.url}/${cartId}`);
+  }
+
+  update(cartId: DaffCart['id'], address: DaffCartAddress): Observable<Partial<DaffCart>> {
```
Comment (lderrickable) [inline libs/cart/testing/src/drivers/in-memory/cart-shipping-address/cart-shipping-address.service.ts]: this should take `address: Partial<DaffCartAddress>`

File: libs/cart/testing/src/drivers/in-memory/cart-shipping-information/cart-shipping-information.service.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export class DaffInMemoryCartShippingInformationService implements DaffCartShippingInformationServiceInterface<DaffCartShippingRate, DaffCart> {
+  url = '/api/cart-shipping-information';
+
+  constructor(private http: HttpClient) {}
+
+  get(cartId: DaffCart['id']): Observable<DaffCartShippingRate> {
+    return this.http.get<DaffCartShippingRate>(`${this.url}/${cartId}`);
+  }
+
+  update(cartId: DaffCart['id'], info: Partial<DaffCartShippingRate>): Observable<DaffCart> {
```
Comment (lderrickable) [inline libs/cart/testing/src/drivers/in-memory/cart-shipping-information/cart-shipping-information.service.ts]: This should return an `Observable<Partial<DaffCart>>`

File: libs/cart/testing/src/drivers/in-memory/cart-shipping-address/cart-shipping-address.service.ts (hunk 1 above)
Comment (griest024) [reply libs/cart/testing/src/drivers/in-memory/cart-shipping-address/cart-shipping-address.service.ts]: Changed

File: libs/cart/testing/src/drivers/in-memory/cart-shipping-information/cart-shipping-information.service.ts (hunk 2 above)
Comment (griest024) [reply libs/cart/testing/src/drivers/in-memory/cart-shipping-information/cart-shipping-information.service.ts]: Changed

**Pull request merged**